src/components/TextInput/TextInput.jsx

```diff
@@ -57,9 +58,10 @@ const TextInput = ({
   const inputProps = {
     'aria-required': isRequired,
     'aria-label': label || name,
+    'aria-invalid': !!error,
     autoComplete: !autoComplete ? 'off' : autoComplete,
     autoFocus,
-    className: inputClasses,
+    className: error ? 'error' : null,
```

Comment:
@nathanyoung the error class should probably be added to the inputClasses object above so we don't lose any custom classes applied by the library user.

something like:
```
const inputClasses = classNames(
    className,
    'TextInput',
    { 'error': error },
  );


const inputProps = {
...
className: inputClasses,
...
};
```